charms/kfp-api/src/charm.py

```diff
@@ -289,40 +289,50 @@ def _get_interfaces(self):
         return interfaces
 
     def _get_mysql(self):
-        mysql = self.model.relations["mysql"]
-        if len(mysql) == 0:
-            raise CheckFailedError("Missing required relation for mysql", BlockedStatus)
-        elif len(mysql) > 1:
-            raise CheckFailedError("Too many mysql relations", BlockedStatus)
+        """Returns mysql relation data from the relation with a mysql database.
 
-        try:
-            mysql = mysql[0]
-            unit = list(mysql.units)[0]
-            mysql = mysql.data[unit]
-        except Exception as e:
-            self.log.exception(
-                f"Encountered the following exception when parsing mysql relation: " f"{str(e)}"
-            )
-            raise CheckFailedError(
-                "Unexpected error when parsing mysql relation.  See logs", BlockedStatus
-            )
+        Raises:
+            CheckFailed(..., BlockedStatus) if there is no mysql relation
+            CheckFailed(..., WaitingStatus) if The remove unit has not joined the relation
+            CheckFailed(..., WaitingStatus) if the relation data bag is empty
+        """
+        mysql_relation = self.model.get_relation("mysql")
 
-        expected_attributes = ["database", "host", "root_password", "port"]
+        # Raise exception and stop execution if the mysql relation is not established
+        if not mysql_relation:
+            raise CheckFailedError("There is no mysql relation", BlockedStatus)
```

Comment:
maybe ? more actionable. But up to you
```suggestion
            raise CheckFailedError("Add mysql relation", BlockedStatus)
```